setup.py

```diff
@@ -1,30 +1,63 @@
-import glob
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, immuneML Development Team.
+# Distributed under the LGPLv3 License. See LICENSE for more info.
+"""immuneML.
+
+This file is part of immneML.
 
+immuneML is free software, please our term and conditions detailed
+in the file LIVENCE.md
+"""
+import ast
+import glob
+import pathlib
+from codecs import open as openc
 from setuptools import setup, find_packages
 
-from source.environment.Constants import Constants
+FULL_VERSION = '0.9.0.dev0'  # Automatically set by setup_version.py
+
+
+def get_long_description():
+    """Return the contents of README.md as a string."""
+    here = pathlib.Path(__file__).absolute().parent
+    long_description = ''
+    with openc(here.joinpath('README.md'), encoding='utf-8') as fileh:
+        long_description = fileh.read()
+    return long_description
+
+
+def get_version():
+    """Return the version from version.py as a string."""
+    here = pathlib.Path(__file__).absolute().parent
+    filename = here.joinpath('source', 'version.py')
+    with openc(filename, encoding='utf-8') as fileh:
+        for lines in fileh:
+            if lines.startswith('FULL_VERSION ='):
+                version = ast.literal_eval(lines.split('=')[1].strip())
+                return version
+    return FULL_VERSION
 
 
 def import_requirements(filename) -> list:
+    """Import requirements."""
     with open(filename, 'r') as file:
         requirements = file.read().split("\n")
     return requirements
 
 
 setup(
-    name="immune-ml",
-    version=Constants.VERSION,
+    name="immuneML",
+    version=get_version(),
     description="immuneML is a software platform for machine learning analysis of immune receptor sequences",
-    long_description=open("README.md").read(),
+    long_description=get_long_description(),
     author="Milena Pavlovic",
     author_email="milenpa@student.matnat.uio.no",
     url="https://github.com/uio-bmi/ImmuneML",
     install_requires=import_requirements("requirements.txt"),
     extras_require={
         "R_plots":  import_requirements("requirements_R_plots.txt"),
-        "DeepRC":  ["widis-lstm-tools@git+https://github.com/widmi/widis-lstm-tools", "deeprc@git+https://github.com/ml-jku/DeepRC@fec4b4f4b2cd70e00e8de83da169560dec73a419"],
         "TCRDist": import_requirements("requirements_TCRdist.txt"),
-        "all": ["rpy2", "widis-lstm-tools@git+https://github.com/widmi/widis-lstm-tools", "deeprc@git+https://github.com/ml-jku/DeepRC@fec4b4f4b2cd70e00e8de83da169560dec73a419", "parasail==1.2", "tcrdist3>=0.1.6"],
```

Comment:
We need to keep this option 'all' (it should just contain the dependencies of all other 'extras') for things such as galaxy, so we do not need to update the extras there every time it changes

Reply:
I have to see how to do it then, because pypy and pip have their restrictions... 

pipy needs to have a certain syntax
pip (after 19, as in the docker) does not accept to have, in the setup, links to packages in git. They recommended (enforce actually) to have users add explicitly their installation. I gave a look to the various options, nothing really nice came out